connector/src/test/java/com/google/cloud/spark/bigquery/v2/BigQueryEstimatedTableStatisticsTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery.v2;
+
+import com.google.cloud.bigquery.*;
+import com.google.cloud.bigquery.connector.common.BigQueryClient;
+import com.google.cloud.bigquery.connector.common.TableStatistics;
+import com.google.common.collect.ImmutableList;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BigQueryEstimatedTableStatisticsTest {
+
+  public static final TableId TABLE_ID = TableId.of("dataset", "table");
+  public static final StandardTableDefinition TABLE_DEFINITION =
+      StandardTableDefinition.newBuilder().setNumBytes(1000L).setNumRows(10L).build();
+  public static final TableInfo TABLE = mock(TableInfo.class);
+
+  @BeforeClass
+  public static void prepareMocks() {
+    when(TABLE.getNumBytes()).thenReturn(TABLE_DEFINITION.getNumBytes());
+    when(TABLE.getNumRows()).thenReturn(BigInteger.valueOf(TABLE_DEFINITION.getNumRows()));
+    when(TABLE.getDefinition()).thenReturn(TABLE_DEFINITION);
+  }
+
+  @Test
+  public void testNoFilterNoProjection() {
+    BigQueryClient bigQueryClient = mock(BigQueryClient.class);
+    prepareCalculateSizeNoFilter(bigQueryClient);
+    BigQueryEstimatedTableStatistics s =
+        BigQueryEstimatedTableStatistics.newBuilder(TABLE, bigQueryClient).build();
+    assertThat(s.numRows().isPresent()).isTrue();
```

Comment:
might be nice to adapt Truth's [Optional Matcher](https://github.com/google/truth/blob/master/extensions/java8/src/main/java/com/google/common/truth/OptionalSubject.java) to a new matcher that works with OptionalLong